Modules/_testinternalcapi/set.c

```diff
@@ -0,0 +1,57 @@
+#include "parts.h"
+#include "../_testcapi/util.h"  // NULLABLE, RETURN_INT
+
+#include "pycore_setobject.h"
+
+
+static PyObject *
+set_update(PyObject *self, PyObject *args)
+{
+    PyObject *set, *iterable;
+    if (!PyArg_ParseTuple(args, "OO", &set, &iterable)) {
+        return NULL;
+    }
+    NULLABLE(set);
+    NULLABLE(iterable);
+    RETURN_INT(_PySet_Update(set, iterable));
+}
+
+static PyObject *
+set_next_entry(PyObject *self, PyObject *args)
+{
+    int rc;
+    Py_ssize_t pos;
+    Py_hash_t hash;
+    PyObject *set, *item = UNINITIALIZED_PTR;
+    if (!PyArg_ParseTuple(args, "On", &set, &pos)) {
+        return NULL;
+    }
+    NULLABLE(set);
+
+    rc = _PySet_NextEntry(set, &pos, &item, &hash);
+    if (rc == 1) {
+        return Py_BuildValue("innO", rc, pos, hash, item);
+    }
+    assert(item == UNINITIALIZED_PTR);
+    if (PyErr_Occurred()) {
```

Comment:
```suggestion
    if (rc == -1) {
```
Otherwise returning -1 without setting error will pass unnoticing.